apps/files_sharing/lib/Command/FixBrokenShares.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Files_Sharing\Command;
+
+use OC\Core\Command\Base;
+use OCA\Files_Sharing\OrphanHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class FixBrokenShares extends Base {
+	public function __construct(
+		private OrphanHelper $orphanHelper
+	) {
+		parent::__construct();
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('sharing:fix-broken-shares')
+			->setDescription('Fix broken shares after transfer ownership')
+			->addOption(
+				'dry-run',
+				null,
+				InputOption::VALUE_NONE,
+				'only show which shares would be updated'
+			);
+	}
+
+	public function execute(InputInterface $input, OutputInterface $output): int {
+		$shares = $this->orphanHelper->getAllShares();
+		$dryRun = $input->getOption('dry-run');
+
+		foreach ($shares as $share) {
+			if ($this->orphanHelper->isShareValid($share['owner'], $share['fileid']) || !$this->orphanHelper->fileExists($share['fileid'])) {
+				continue;
+			}
+
+			$owner = $this->orphanHelper->findOwner($share['fileid']);
+
+			if (!empty($owner)) {
```

Comment:
```suggestion
			if ($owner !== null) {
```
`'0'` is a valid userid.